source/features/remove-unused-repo-tabs.tsx

```diff
@@ -0,0 +1,106 @@
+import cache from 'webext-storage-cache';
+import select from 'select-dom';
+import elementReady from 'element-ready';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import fetchDom from '../helpers/fetch-dom';
+import getTabCount from '../github-helpers/get-tab-count';
+import looseParseInt from '../helpers/loose-parse-int';
+import {getWorkflows} from './next-scheduled-github-action';
+import abbreviateNumber from '../helpers/abbreviate-number';
+import {onlyShowInDropdown} from './more-dropdown';
+import {buildRepoURL, getRepo} from '../github-helpers';
+
+function mustKeepTab(tab: HTMLElement | undefined): boolean {
+	return (
+		!tab || // Tab disabled 🎉
+		tab.matches('.selected') ||// User is on tab 👀
+		// Repo/Organization owners should see the tab. If they don't need it, they should disable the feature altogether
+		pageDetect.canUserEditRepo() ||
+		pageDetect.canUserEditOrganization()
+	);
+}
+
+function setTabCounter(tab: HTMLElement, count: number): void {
+	const tabCounter = select('.Counter', tab)!;
+	tabCounter.textContent = abbreviateNumber(count);
+	tabCounter.title = count > 999 ? String(count) : '';
+}
+
+const getWikiPageCount = cache.function(async (): Promise<number> => {
+	const wikiPages = await fetchDom(buildRepoURL('wiki'), '#wiki-pages-box .Counter');
+	return looseParseInt(wikiPages);
+}, {
+	maxAge: {hours: 1},
+	staleWhileRevalidate: {days: 5},
+	cacheKey: () => 'wiki-page-count:' + getRepo()!.nameWithOwner,
+});
+
+async function initWiki(): Promise<void | false> {
+	const wikiTab = await elementReady('[data-hotkey="g w"]');
+	if (!wikiTab) {
+		return false;
+	}
+
+	const wikiPageCount = await getWikiPageCount();
+	if (wikiPageCount > 0 || mustKeepTab(wikiTab)) {
+		setTabCounter(wikiTab, wikiPageCount);
+	} else {
+		onlyShowInDropdown('wiki-tab');
+	}
+}
+
+async function initActions(): Promise<void | false> {
+	const actionsTab = await elementReady('[data-hotkey="g a"]');
+	if (!actionsTab) {
+		return false;
+	}
+
+	const actionsCount = (await getWorkflows()).length;
+	if (actionsCount > 0 || mustKeepTab(actionsTab)) {
+		setTabCounter(actionsTab, actionsCount);
+	} else {
+		onlyShowInDropdown('actions-tab');
+	}
+}
+
+async function initProjects(): Promise<void | false> {
+	const projectsTab = await elementReady('[data-hotkey="g b"]');
+	if (await getTabCount(projectsTab!) > 0 || mustKeepTab(projectsTab)) {
+		return false;
+	}
+
+	if (pageDetect.isRepo()) {
+		onlyShowInDropdown('projects-tab');
+		return;
+	}
+
+	// .isOrganizationProfile
+	if (await elementReady('a.btn-primary[href$="repositories/new"]')) {
```

Comment:
I'd explain what this is though, the comment doesn't help me understand what it does and why we have it

Reply:
```suggestion
	if (await elementReady('a.btn-primary[href$="repositories/new"]')) { // Check if the user is able to create a new repository
```

Good?

Reply:
Yes but why are we checking that? (This need to be in the comment, not here)

Reply:
In short, something like: “only remove the tab if you’re not a collaborator”

It may be easier to extract this into its own local function: canUserEditWhateverThisPageIs()

And then comment on its use as “Leave tab visible to allow the current user to create Projects”

Reply:
Also there’s one issue specific to negative usage of elementReady: When used this way, you’re basically just waiting for domReady and then removing the tab. It’s still ok but it’d be best to use:

```js 
elementReady(element that always exists)?.querySelector(the button)
``` 

The difference is that this will resolve sooner when the button doesn’t exist

Reply:
uhmmm there's already a `pageDetect.canUserEditOrganization`, so what is this line for exactly?